run.py

```diff
@@ -6,20 +6,28 @@
 import pdb
 
 from uuid_encoder import UUIDEncoder
-from uuid_validator import UUIDValidator
+from custom_validator import CustomValidator
 from eve import Eve
 from flask import request, jsonify, abort, Response, current_app
 from flask_bootstrap import Bootstrap
 from eve_swagger import swagger
 from bson import json_util
 from flask_zipkin import Zipkin
+from pymongo import ReturnDocument
 
 environment = os.getenv('EVE_ENV', 'development')
 
 SETTINGS_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'db', environment + '.py')
 
 def create_app(settings):
-  app = Eve(settings=settings, json_encoder=UUIDEncoder, validator=UUIDValidator)
+  app = Eve(settings=settings, json_encoder=UUIDEncoder, validator=CustomValidator)
+
+  # We are using a document in the counters collection to generate sequential ids to be
+  # used for barcodes. Here we're "seeding" the collection with the inital document
+  with app.app_context():
+    current_app.data.driver.db \
+      .get_collection('counters') \
+      .update({ '_id': 'barcode' }, { 'seq': 0 }, upsert=True )
```

Comment:
Okay, looks like if we change line 30 to

`.update({ '_id': 'barcode' }, { '$setOnInsert': { 'seq': 0 } }, upsert=True )`

should work as we want it to.